examples/deploy_contract/main.go

```diff
@@ -93,18 +93,20 @@ func DeployContractDemo() {
 		[]templates.Contract{{
 			Name:   "GreatToken",
 			Source: nftCode,
-		}}, myAddress)
+		}}, serviceAcctAddr)
 
-	deployContractTx.SetProposalKey(myAddress, myAcctKey.Index, myAcctKey.SequenceNumber)
+	deployContractTx.SetProposalKey(
+		serviceAcctAddr,
+		serviceAcctKey.Index,
+		serviceAcctKey.SequenceNumber,
+	)
 	// we can set the same reference block id. We shouldn't be to far away from it
 	deployContractTx.SetReferenceBlockID(referenceBlockID)
-	deployContractTx.SetPayer(myAddress)
+	deployContractTx.SetPayer(serviceAcctAddr)
 
-	err = deployContractTx.SignEnvelope(
-		myAddress,
-		myAcctKey.Index,
-		crypto.NewInMemorySigner(myPrivateKey, myAcctKey.HashAlgo),
```

Comment:
Its a bit tricky actually! if minimum account balance is on, the newly created has 0.1 FLOW. If this account tries to create an account it needs 0.1 FLOW, but it cant use its own flow for that, so the second transaction fails. 

The solution is to add FLOW to the account, but there is no example that uses (or knows about) the FungibleToken address. I wasn't sure how to approach that.

I think the examples should have an `examples.addFLOWToAddress` method which calls a minting transaction, but I need to hardcode the FungibleToken and FlowToken addresses. Do you think that is ok?

Reply:
Hmm, I don't think it's simpler if we just assume that the account that attempts to deploy the contract has enough FLOW / prepared the target account, so that it has enough storage capacity.
We could just add a note/comment at the top of the example that points out this assumption, maybe something like:

```
// NOTE: This example assumes that the target account has enough storage capacity to store the source code and instantiated contract
``` 

I'm worried that the use of the service account, or minting and adding FLOW, will confuse users/developers, because both aren't things that they are able to do.

Reply:
Ok let me give you slightly more detail, cause I suspect we might not be 100% on the same page:

That example does (S -service account, A# - other accounts):
1. S creates A1
2. A1 creates A2
3. ... (this isn't relevant for now)

With storage fees this means:
1. S creates A1 and adds the minimum deposit (0.1 FLOW) to A1 
2. A1 creates A2 and adds the minimum deposit to A2, but this means A1 doesn't have the required minimum anymore! So this TX fails

This will always happen (regardless of how large the contract is that is being deployed), unless A1 has more than the minimum deposit.

Reply:
Sounds good! I just wanted to make sure we don't over complicate the simple example of deploying a contract